csi/controllerserver.py

```diff
@@ -255,6 +258,133 @@ def ControllerGetCapabilities(self, request, context):
+        #             pvtype = PV_TYPE_VIRTBLOCK
+
+        pvtype = PV_TYPE_SUBVOL
+
+        single_node_writer = getattr(csi_pb2.VolumeCapability.AccessMode,
+                                             "SINGLE_NODE_WRITER")
+
+        if request.volume_capability.AccessMode == single_node_writer:
+            pvtype = PV_TYPE_VIRTBLOCK
+
+        logging.debug(logf(
+            "Found PV type",
+            pvtype=pvtype,
+            capability=request.volume_capability
+        ))
+
+        # Add everything from parameter as filter item
+        # filters = {}
+        # for pkey, pvalue in request.parameters.items():
+        #     filters[pkey] = pvalue
+
+        # logging.debug(logf(
+        #     "Filters applied to choose storage",
+        #     **filters
+        # ))
+
+        # # UID is stored at the time of installation in configmap.
+        # uid = None
+        # with open(os.path.join(VOLINFO_DIR, "uid")) as uid_file:
+        #     uid = uid_file.read()
+
+        # host_volumes = get_pv_hosting_volumes(filters)
+        # logging.debug(logf(
+        #     "Got list of hosting Volumes",
+        #     volumes=",".join(v['name'] for v in host_volumes)
+        # ))
+
+        # hostvol = mount_and_select_hosting_volume(host_volumes, expansion_requested_pvsize)
+        # if hostvol is None:
+        #     errmsg = "No Hosting Volumes available, add more storage"
+        #     logging.error(errmsg)
+        #     context.set_details(errmsg)
+        #     context.set_code(grpc.StatusCode.RESOURCE_EXHAUSTED)
+        #     return csi_pb2.CreateVolumeResponse()
+
+        # mntdir = os.path.join(HOSTVOL_MOUNTDIR, hostvol)
+
+        # if request.vol_capability.BlockVolume:
+        #     logging.info(logf(
+        #         "Volume is of type 'block'"
+        #     ))
+        #     expand_volume(mntdir)
+
+        # if pvtype == PV_TYPE_VIRTBLOCK:
+        #     vol = update_virtblock_volume(
+        #         mntdir, request.name, expansion_requested_pvsize)
+        # else:
+        #     vol = update_subdir_volume(
+        #         mntdir, request.name, expansion_requested_pvsize)
+
+        # logging.info(logf(
+        #     "Volume expanded",
+        #     name=request.name,
+        #     size=expansion_requested_pvsize,
+        #     hostvol=hostvol,
+        #     pvtype=pvtype,
+        #     volpath=vol.volpath,
+        #     duration_seconds=time.time() - start_time
+        # ))
+
+        # sizechanged is the additional change to be
+        # subtracted from storage-pool
+        # sizechange = expansion_requested_pvsize - existing_pvsize
+        # update_free_size(hostvol, request.name, -sizechange)
+
+        # if not hostvoltype:
+        #    hostvoltype = "unknown"
+
+        # send_analytics_tracker("pvc-%s" % hostvoltype, uid)
+        return csi_pb2.ControllerExpandVolumeResponse({
+                "capacity_bytes": expansion_requested_pvsize,
```

Comment:
I think the response message is wrong, I'm not getting the right format. Below is the prescribed format
```
message ControllerExpandVolumeResponse {
  // Capacity of volume after expansion. This field is REQUIRED.
  int64 capacity_bytes = 1;

  // Whether node expansion is required for the volume. When true
  // the CO MUST make NodeExpandVolume RPC call on the node. This field
  // is REQUIRED.
  bool node_expansion_required = 2;
}
``` 
I'm getting error 
```
 return csi_pb2.ControllerExpandVolumeResponse({
TypeError: No positional arguments allowed
```

Reply:
Nope, don't pass anything here. That is handled by csi_pb2.py file.